packages/camera/example/lib/main.dart

```diff
@@ -24,149 +26,315 @@ IconData cameraLensIcon(CameraLensDirection direction) {
   throw new ArgumentError('Unknown lens direction');
 }
 
+void logError(String code, String message) =>
+    print('Error: $code\nError Message: $message');
+
 class _CameraExampleHomeState extends State<CameraExampleHome> {
-  bool opening = false;
   CameraController controller;
   String imagePath;
-  int pictureCount = 0;
+  String videoPath;
+  VideoPlayerController videoController;
+  VoidCallback videoPlayerListener;
 
-  @override
-  void initState() {
-    super.initState();
-  }
+  final GlobalKey<ScaffoldState> _scaffoldKey = new GlobalKey<ScaffoldState>();
 
   @override
   Widget build(BuildContext context) {
-    final List<Widget> headerChildren = <Widget>[];
+    return new Scaffold(
+      key: _scaffoldKey,
+      appBar: new AppBar(
+        title: const Text('Camera example'),
+      ),
+      body: new Column(children: <Widget>[
+        new Expanded(
+          child: new Container(
+            child: new Padding(
+              padding: const EdgeInsets.all(1.0),
+              child: new Center(
+                child: _cameraPreviewWidget(),
+              ),
+            ),
+            decoration: new BoxDecoration(
+              color: Colors.black,
+              border: new Border.all(
+                color: controller != null && controller.value.isRecordingVideo
+                    ? Colors.redAccent
+                    : Colors.grey,
+                width: 3.0,
+              ),
+            ),
+          ),
+        ),
+        new Padding(
+          padding: const EdgeInsets.all(5.0),
+          child: new Row(
+              mainAxisAlignment: MainAxisAlignment.start,
+              children: <Widget>[
+                _cameraTogglesRowWidget(),
+                _thumbnailWidget(),
+              ]),
+        ),
+        _captureControlRowWidget(),
```

Comment:
@sigurdm  It's indeed annoying the snackbar is covering the controls. Do you know if there is any way to have the snackbar come from the top of the screen instead of the bottom?

Reply:
No unfortunately. I think the material spec says it has to come from the bottom. (https://material.io/guidelines/components/snackbars-toasts.html#snackbars-toasts-usage)
Would it work to have the controls at the top?

Reply:
I think it would be a bit awkward to have to reach this high (and put you hand in front of the preview). We will move the row up as you suggested, that's already an improvement. We would need an actual designer to look into the app to make something good anyway :sweat_smile: